samples/core/guide/autograph.ipynb

```diff
@@ -609,6 +646,217 @@
       "execution_count": 0,
       "outputs": []
     },
+    {
+      "metadata": {
+        "id": "XY4UspHmZNdL",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Interoperation with `tf.Keras`\n",
+        "\n",
+        "Now that you've seen the basics, let's build some real model components with autograph.\n",
+        "\n",
+        "It's relatively simple to integrate `autograph` with `tf.keras`. But remember that batchng is essential for performance. So the best candidate code for conversion to autograph is code where the control flow is decided at the _batch_ level. If decisions are made at the individual _example_ level you will still need to index and batch your examples to maintain performance while appling the control flow logic. \n",
+        "\n",
+        "\n",
+        "### Stateless functions\n",
+        "\n",
+        "For stateless functions like `collatz`, below, the easiest way to include them in a keras model is to wrap them up as a layer uisng `tf.keras.layers.Lambda`."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "ChZh3q-zcF6C",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "import numpy as np\n",
+        "\n",
+        "@autograph.convert()\n",
+        "def collatz(x):\n",
+        "  x=tf.reshape(x,())\n",
+        "  assert x>0\n",
+        "  n = tf.convert_to_tensor((0,)) \n",
+        "  while not tf.equal(x,1):\n",
+        "    n+=1\n",
+        "    if tf.equal(x%2, 0):\n",
+        "      x = x//2\n",
+        "    else:\n",
+        "      x = 3*x+1\n",
+        "      \n",
+        "  return n\n",
+        "\n",
+        "with tf.Graph().as_default():\n",
+        "  model = tf.keras.Sequential([\n",
+        "    tf.keras.layers.Lambda(collatz, input_shape=(1,), output_shape=(), )\n",
+        "  ])\n",
+        "  \n",
+        "result = model.predict(np.array([6171])) #261\n",
+        "result"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "k9LEoa3ud9hA",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Custom Layers and Models\n",
+        "\n",
+        "<!--TODO(markdaoust) link to full examples  or these referenced models.-->\n",
```

Comment:
TODO now or later?

Reply:
When I can show that this performs as the paper: https://colab.research.google.com/drive/14YkY78YjBhqyKn9pQ50rC36hWkuWHFb7#scrollTo=nzst0_CH2uXV